coordinator/src/main/scala/filodb.coordinator/TenantIngestionMetering.scala

```diff
@@ -0,0 +1,89 @@
+package filodb.coordinator
+
+import java.util.concurrent.TimeUnit
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.{Failure, Success}
+
+import akka.actor.ActorRef
+import com.typesafe.scalalogging.StrictLogging
+import kamon.Kamon
+import kamon.tag.TagSet
+import monix.execution.Scheduler.Implicits.{global => scheduler}
+
+import filodb.coordinator.client.Client
+import filodb.coordinator.client.QueryCommands.LogicalPlan2Query
+import filodb.core.DatasetRef
+import filodb.query.{QueryError, QueryResult, TsCardinalities}
+
+/**
+ * Periodically queries a node for all namespace cardinalities.
+ * Kamon gauges are updated with the response data.
+ *
+ * The intent is to publish a low-cardinality metric such that namespace
+ * cardinality queries can be efficiently answered.
+ *
+ * @param dsIterProducer produces an iterator to step through all datasets.
+ * @param coordActorProducer produces a single actor to ask a query. Actors are
+ *          queried in the order they're returned from this function.
+ */
+case class TenantIngestionMetering(settings: FilodbSettings,
+                                   dsIterProducer: () => Iterator[DatasetRef],
+                                   coordActorProducer: () => ActorRef) extends StrictLogging{
+
+  private val ASK_TIMEOUT = FiniteDuration(
+    settings.config.getDuration("shard-manager.metering-query-interval").toSeconds,
+    TimeUnit.SECONDS)
+  private val SCHED_INIT_DELAY = ASK_TIMEOUT  // time until first job is scheduled
+  private val SCHED_DELAY = ASK_TIMEOUT  // time between all jobs after the first
```

Comment:
Let's call out a TODO to fix this in different PR. If for whatever reason the query takes longer than timeout consistently, metric may never be published. Instead, the we could slow down the queries and publish less often. Need not be done in this PR.

Reply:
Noted-- I'll define a throttling function in my next PR.

Reply:
Included in #1310.